benchmarks/allocation_size_benchmarks.cpp

```diff
@@ -0,0 +1,206 @@
+      deallocate(m_allocations[j]);
+  }
+
+  void deallocation(benchmark::State& st) {
+    const std::size_t size{static_cast<std::size_t>(st.range(0))};
+    std::size_t i{0};
+
+    while (st.KeepRunning()) {
+      if (i == 0 || i == Max_Allocations) {
+        st.PauseTiming();
+        for (std::size_t j{0}; j < Max_Allocations; j++)
+          m_allocations[j] = allocate(size);
+        i = 0;
+        st.ResumeTiming();
+      }
+      deallocate(m_allocations[i++]);
+    }
+    for (std::size_t j{i}; j < Max_Allocations; j++)
+      deallocate(m_allocations[j]);
+  }
+
+  void* m_allocations[Max_Allocations];
+};
+
+template <typename Alloc>
+class ResourceAllocator : public AllocatorBenchmark
+{
+public:
+  ResourceAllocator() : m_alloc{} {}
+  virtual void* allocate(std::size_t nbytes) final { return m_alloc.allocate(nbytes); }
+  virtual void deallocate(void* ptr) final { m_alloc.deallocate(ptr); }
+private:
+  Alloc m_alloc;
+};
+
+class Malloc : public ResourceAllocator<umpire::alloc::MallocAllocator> {};
+BENCHMARK_DEFINE_F(Malloc, malloc)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(Malloc, free)(benchmark::State& st)   { deallocation(st); }
+
+#if defined(UMPIRE_ENABLE_CUDA)
+class CudaMalloc : public ResourceAllocator<umpire::alloc::CudaMallocAllocator> {};
+BENCHMARK_DEFINE_F(CudaMalloc, cudaMalloc)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(CudaMalloc, cudaFree)(benchmark::State& st)   { deallocation(st); }
+
+class CudaMallocManaged : public ResourceAllocator<umpire::alloc::CudaMallocManagedAllocator> {};
+BENCHMARK_DEFINE_F(CudaMallocManaged, cudaMallocManaged)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(CudaMallocManaged, cudaFree)(benchmark::State& st)   { deallocation(st); }
+
+class CudaPinned : public ResourceAllocator<umpire::alloc::CudaPinnedAllocator> {};
+BENCHMARK_DEFINE_F(CudaPinned, cudaMallocHost)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(CudaPinned, cudaFreeHost)(benchmark::State& st)   { deallocation(st); }
+#endif
+
+#if defined(UMPIRE_ENABLE_HIP)
+class HipMalloc : public ResourceAllocator<umpire::alloc::HipMallocAllocator> {};
+BENCHMARK_DEFINE_F(HipMalloc, hipMalloc)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(HipMalloc, hipFree)(benchmark::State& st)   { deallocation(st); }
+
+class HipMallocManaged : public ResourceAllocator<umpire::alloc::HipMallocManagedAllocator> {};
+BENCHMARK_DEFINE_F(HipMallocManaged, hipMallocManaged)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(HipMallocManaged, hipFree)(benchmark::State& st)   { deallocation(st); }
+
+class HipPinned : public ResourceAllocator<umpire::alloc::HipPinnedAllocator> {};
+BENCHMARK_DEFINE_F(HipPinned, hipMallocHost)(benchmark::State& st) { allocation(st); }
+BENCHMARK_DEFINE_F(HipPinned, hipFreeHost)(benchmark::State& st)   { deallocation(st); }
+#endif
+
+template <umpire::resource::MemoryResourceType Resource>
```

Comment:
We don't need these `MemoryResourceAllocator` benchmarks in this file.

Reply:
Gasp! Then are we keeping those tests at all? (Because the 3rd benchmark is only for pools)

Reply:
I think the value has gone since they were used to compare overhead, but we now have the no-op benchmark for that.